exporter/syslogexporter/factory.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syslogexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/syslogexporter"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr        = "syslog"
+	stabilityLevel = component.StabilityLevelAlpha
+)
+
+// NewFactory returns a new factory for the syslog exporter.
+func NewFactory() exporter.Factory {
+	return exporter.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		exporter.WithLogs(createLogsExporter, stabilityLevel),
+	)
+}
+
+func createDefaultConfig() component.Config {
+	qs := exporterhelper.NewDefaultQueueSettings()
+	qs.Enabled = false
+
+	return &Config{
```

Comment:
This is not related to this pull request specifically, but I stumbled upon the fact that the TLS settings apparently do not have a `NewDefaultSettings` method similar to `NewDefaultRetrySettings` or `NewDefaultQueueSettings`. Perhaps it's worth adding such a method on the [TLSSetting](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.73.0/config/configtls/configtls.go) struct?

This is not a problem now, because the default TLS config is equal to an empty, non-initialized struct. But if one day this changes, it would be useful to have the `NewDefaultSettings` method as a central place to change this.

I wonder what the @open-telemetry/collector-approvers think.